server/models/procedures/stock.sql

```diff
@@ -34,10 +34,137 @@ CREATE PROCEDURE ComputeMovementReference (
 )
 BEGIN
   DECLARE reference INT(11);
+  DECLARE flux INT(11);
 
   SET reference = (SELECT COUNT(DISTINCT document_uuid) AS total FROM stock_movement LIMIT 1);
+  SET flux = (SELECT flux_id FROM stock_movement WHERE document_uuid = documentUuid LIMIT 1);
 
   INSERT INTO `document_map` (uuid, text)
-  VALUES (documentUuid, CONCAT('MVT.', reference))
+  VALUES (documentUuid, CONCAT_WS('.', 'SM', flux, reference))
   ON DUPLICATE KEY UPDATE uuid = uuid;
 END $$
+
+-- post stock movement into vouchers
+DROP PROCEDURE IF EXISTS PostStockMovement;
+CREATE PROCEDURE PostStockMovement (
+  IN documentUuid BINARY(16),
+  IN isExit TINYINT(1),
+  IN projectId SMALLINT(5),
+  IN currencyId SMALLINT(5)
+)
+BEGIN
+  -- voucher
+  DECLARE voucher_uuid BINARY(16);
+  DECLARE voucher_date DATETIME;
+  DECLARE voucher_project_id SMALLINT(5);
+  DECLARE voucher_currency_id SMALLINT(5);
+  DECLARE voucher_user_id SMALLINT(5);
+  DECLARE voucher_type_id SMALLINT(3);
+  DECLARE voucher_description TEXT;
+  DECLARE voucher_amount DECIMAL(19, 4);
+
+  -- voucher item
+  DECLARE voucher_item_uuid BINARY(16);
+  DECLARE voucher_item_account INT(10);
+  DECLARE voucher_item_account_debit DECIMAL(19, 4);
+  DECLARE voucher_item_account_credit DECIMAL(19, 4);
+  DECLARE voucher_item_voucher_uuid BINARY(16);
+  DECLARE voucher_item_document_uuid BINARY(16);
+  
+  -- variables
+	DECLARE v_stock_account INT(10);
+	DECLARE v_cogs_account INT(10);
+	DECLARE v_unit_cost DECIMAL(19, 4);
+	DECLARE v_quantity INT(11);
+	DECLARE v_document_uuid BINARY(16);
+	DECLARE v_is_exit TINYINT(1);
+
+  -- transaction type 
+  DECLARE STOCK_EXIT_TYPE SMALLINT(5) DEFAULT 13;
+  DECLARE STOCK_ENTRY_TYPE SMALLINT(5) DEFAULT 14;
+
+
+  -- the cursor 
+  DECLARE v_finished INTEGER DEFAULT 0;
+  
+  DECLARE stage_stock_movement_cursor CURSOR FOR 
+  	SELECT temp.stock_account, temp.cogs_account, temp.unit_cost, temp.quantity, temp.document_uuid, temp.is_exit 
+	FROM stage_stock_movement as temp;
+  
+  DECLARE CONTINUE HANDLER FOR NOT FOUND SET v_finished = 1;
+
+  -- temporarise the stock movement
+  CREATE TEMPORARY TABLE stage_stock_movement (
+      SELECT 
+        projectId as project_id, currencyId as currency_id,
+        m.uuid, m.description, m.date, m.flux_id, m.is_exit, m.document_uuid, m.quantity, m.unit_cost, m.user_id,
+        ig.cogs_account, ig.stock_account 
+      FROM stock_movement m  
+      JOIN lot l ON l.uuid = m.lot_uuid 
+      JOIN inventory i ON i.uuid = l.inventory_uuid 
+      JOIN inventory_group ig 
+        ON ig.uuid = i.group_uuid AND (ig.stock_account IS NOT NULL AND ig.cogs_account IS NOT NULL) 
+      WHERE m.document_uuid = documentUuid AND m.is_exit = isExit
+    );
+
+  -- define voucher variables
+  SET voucher_uuid = (SELECT HUID(UUID()));
```

Comment:
All your select are in one table, instead of having many select like this you can do
``` SELECT @var1 = column1, @var2 = column2 FROM table```